File: jsp_basic(선생님)/src/main/webapp/js/student.js
// 학생객체 생성을 위한 틀(클래스)
class Student {
  constructor(ssn, name, korean, english, math) {
    // this = {};
    this.ssn = ssn;
    this.name = name;
    this.korean = korean;
    this.english = english;
    this.math = math;
  }

  // 정적(static) 속성 및 메소드 추가
  //static schoolName = "Ezen초등학교";

  getSum() {
    return this.korean + this.english + this.math;
  }

  getAverage() {
    return this.getSum() / 3;
  }

  toString() {
    return `${this.ssn} \t ${this.name} \t ${this.korean} \t ${this.english} \t ${this.math} \t ${this.getSum()} \t ${this.getAverage()}`;
  }
}

export {Student};

File: Java_oop/src/ezen/oop/array/Student.java
package ezen.oop.array;

//	학생정보 추상화/캡슐화
//	마커인터페이스(추상메소드가 없는 멍텅구리 인터페이스)

public class Student implements Cloneable {
	public static final String SCHOOL_NAME = "이젠초등학교";
//	관례대로 상수는 인스턴스보다 위에 
	private int no;
	private String name;
	private int score;
	
//	생성자
	public Student() {}
	public Student(int no, String name, int score) {
		this.no = no;
		this.name = name;
		this.score = score;
	}
	
//	getter & setter 메소드
	public int getNo() {
		return no;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public int getScore() {
		return score;
	}
	public void setScore(int score) {
		this.score = score;
	}

//	학생 모든 정보를 출력하는 기능
	public void printInfo () {
		System.out.println(no + "\t" + name + "\t" + score);
	}
	@Override
	public String toString() {
		return no + "\t" + name + "\t" + score;
	}
	
	@Override
	public boolean equals(Object obj) {
		if(obj instanceof Student) {
		Student student = (Student) obj;
			if(toString().equals(student.toString())) {
				return true;
			}
		}
		return false;
	}
	
	@Override
	public Object clone() throws CloneNotSupportedException {
		return super.clone();
	}
}


